core/trie/trie_test.go

```diff
@@ -298,6 +299,41 @@ func TestMaxTrieHeight(t *testing.T) {
 	})
 }
 
+func TestRoot(t *testing.T) {
+	require.NoError(t, trie.RunOnTempTrie(251, func(tempTrie *trie.Trie) error {
+		t.Run("root on empty trie", func(t *testing.T) {
+			want := new(felt.Felt)
+			got, err := tempTrie.Root()
+			require.NoError(t, err)
+			assert.Equal(t, want, got)
+		})
+
+		t.Run("root after one insertion", func(t *testing.T) {
+			bottom := new(felt.Felt).SetUint64(1)
+			key := new(felt.Felt).SetUint64(1)
+
+			_, err := tempTrie.Put(key, bottom)
+			require.NoError(t, err)
+
+			// Update root. Ensure it matches expectations.
+			want := crypto.Pedersen(bottom, key)
+			// We inserted a leaf, so path length is the height of the trie.
+			// That's felt.Bits-1 = 251.
+			want.Add(want, new(felt.Felt).SetUint64(felt.Bits-1))
+			got, err := tempTrie.Root()
+			require.NoError(t, err)
+			assert.Equal(t, want, got)
+
+			// Recalculating after no changes should give same result.
+			got, err = tempTrie.Root()
+			require.NoError(t, err)
+			assert.Equal(t, want, got)
+		})
+
+		return nil
+	}))
+}
+
```

Comment:
can you explain what is the goal of this test?

Reply:
That `trie.Root` matches the spec for an empty trie and after one insertion. Maybe this is just re-testing the node hashing code though...